### PR TITLE
[VFS] Allow to move file or dir to trash

### PR DESCRIPTION
**Pull request by jinroh**
This PR adds the possibility to move file and directories to the trash. It simply move them on a special trash directory for now. I have not added anything more for now to handle file recovery.

It still requires more tests, but it is ready for review.

### CODE REVIEW

File: vfs/directory.go [hunk 1]
```diff
@@ -363,6 +367,19 @@ func bulkUpdateDocsPath(c Context, oldpath, newpath string) error {
 	return err
 }
 
+// TrashDir is used to delete a directory given its document
+func TrashDir(c Context, olddoc *DirDoc) (newdoc *DirDoc, err error) {
+	trashFolderID := TrashFolderID
+	err = tryOrUseSuffix(olddoc.Name, "%scozy__%s", func(name string) error {
+		newdoc, err = ModifyDirMetadata(c, olddoc, &DocPatch{
```
Comment (nono) [inline vfs/directory.go]: Why do you assign `newdoc` inside the anonymous func but return `err` to assign it from `TrashDir`?

File: vfs/file.go [hunk 2]
```diff
@@ -523,6 +509,19 @@ func ModifyFileMetadata(c Context, olddoc *FileDoc, patch *DocPatch) (newdoc *Fi
 	return
 }
 
+// TrashFile is used to delete a file given its document
+func TrashFile(c Context, olddoc *FileDoc) (newdoc *FileDoc, err error) {
+	trashFolderID := TrashFolderID
+	err = tryOrUseSuffix(olddoc.Name, "%scozy__%s", func(name string) error {
+		newdoc, err = ModifyFileMetadata(c, olddoc, &DocPatch{
```
Comment (nono) [inline vfs/file.go]: Idem

File: vfs/rand_suffix.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		err = do(newname)
+		if os.IsExist(err) {
+			if nconflict++; nconflict > 10 {
+				randmu.Lock()
+				rand = reseed()
+				randmu.Unlock()
+			}
+			continue
+		}
+		break
```
Comment (nono) [inline vfs/rand_suffix.go]: Won't it be simpler to inverse the `if` condition, like this?

```go
for i := 0; i < 1000; i++ {
    // ...
    if !os.IsExist(err) {
        break
    }
    if nconflict++; nconflict > 10 {
        randmu.Lock()
        rand = reseed()
        randmu.Unlock()
    }
}
```

File: web/files/files_test.go [hunk 4]
```diff
@@ -852,6 +869,23 @@ func TestGetDirectoryMetadataFromID(t *testing.T) {
...
+	dirID, _ := extractDirData(t, data1)
+
+	res2, data2 := trash(t, "/files/"+dirID)
+	assert.Equal(t, 200, res2.StatusCode)
+	fmt.Println(data2)
+}
+
+func TestFileTrash(t *testing.T) {
+
+}
```
Comment (nono) [inline web/files/files_test.go]: This test should not fail often ;-)

Comment (jinroh) [inline web/files/files_test.go]: 😄  like said in PR, i need to add more tests (yay !)

File: vfs/vfs.go [hunk 5]
```diff
@@ -34,11 +34,18 @@ const DefaultContentType = "application/octet-stream"
...
+	// FsDocType is document type
+	FsDocType = "io.cozy.files"
+	// RootFolderID is the identifier of the root directory
+	RootFolderID = "io.cozy.files.rootdir"
+	// TrashFolderID is the identifier of the trash directory
+	TrashFolderID = "io.cozy.files.trashdir"
+	// TrashDirName is the path of the trash directory
+	TrashDirName = "/.cozy_trash"
+	// AppsDirName is the path of the directory in which apps are stored
+	AppsDirName = "/.cozy_apps"
```
Comment (nono) [inline vfs/vfs.go]: In the documentation, it's `/Apps`. We should fix the code or the documentation to be coherent

**Pull request merged**